ViAn/Video/video_player.cpp

```diff
@@ -532,3 +578,39 @@ void video_player::export_current_frame(QString path_to_folder) {
     QImageWriter writer(path_to_folder, "tiff");
     writer.write(img);
 }
+
+/**
+ * @brief video_player::scaling_event
+ * This slot gets called when the video QLabel gets resized.
+ * Sets the video width and height to match.
+ * @param width
+ * @param height
+ */
+void video_player::scaling_event(int new_width, int new_height) {
+    if (!capture.isOpened()) {
+        return;
+    }
+
+
+    int video_width = capture.get(CV_CAP_PROP_FRAME_WIDTH);
+    int video_height = capture.get(CV_CAP_PROP_FRAME_HEIGHT);
+    float height_ratio = float(new_height)/float(video_height);
+    float width_ratio = float(new_width)/float(video_width);
+
+    //This statement ensures that the original aspect ratio of the video is kept when scaling
+    if (width_ratio >= height_ratio) {
+        frame_width = int(video_width * height_ratio);
+        frame_height = new_height;
+    } else {
+        frame_width = new_width;
+        frame_height = int(video_height * width_ratio);
+    }
+}
+
+/**
+ * @brief video_player::video_open
+ * @return a bool that is true as long as a video is open.
+ */
+bool video_player::video_open() {
+    return capture.isOpened();
+}

```

Comment:
Maybe this function should be used in the video_player class? Or is it just thought as a getter for the mainwindow to use?